src/App.jsx

```diff
@@ -14,70 +15,81 @@ export const goods = [
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const selectClassName = good => (
+    selectedGood === good
+      ? 'has-background-success-light'
+      : ''
+  );
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  const clearSelectedGood = () => {
+    setSelectedGood('');
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      {selectedGood
+        ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => clearSelectedGood()}
+            />
+          </h1>
+        )
+        : (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )
+      }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={selectClassName(good)}
             >
-              +
-            </button>
-          </td>
+              <td>
+                {selectedGood === good
+                  ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => clearSelectedGood()}
+                    >
+                      -
+                    </button>
+                  )
+                  : (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => setSelectedGood(good)}
```

Comment:
```suggestion
                      onClick={() => selectGood(good)}
```
I'd recommend creating a separate function to handle this logic. Though this function will have similar functionality but this way your code will be more straightforward, and easier to maintain, extend and re-use.